client/testing/test_upload.py

```diff
@@ -111,6 +111,9 @@ def test_vcs_export(self, uploadhub, repo, setupdir, tmpdir, monkeypatch):
             assert newrepo.join(repo.basename).join(".hg").listdir()
         else:
             assert newrepo.join(repo.basename).join(".git").listdir()
+        with uploadhub.workdir() as uploadbase:
+            exported = checkout.export(uploadbase)
```

Comment:
You should make a file read-only here. The test should then fail if you don't add the ``onerror``.

Reply:
Done